pkg/grpc/server.go

```diff
@@ -86,14 +86,20 @@ func (s *server) handler(stream stream, selector cache.ResourceSelector) error {
 	var version string
 
 	for {
+		config := cache.Promise{}
+		defer func() {
+			if config.Stop != nil {
```

Comment:
If config.Value is a method (see below) then this potentially becomes.

defer config.Value() // read and discard value.

Reply:
We don't want to continue waiting for the value if a new version is requested.